lib/rubycritic/cli/application.rb

```diff
@@ -14,12 +14,29 @@ def initialize(argv)
 
       def execute
         parsed_options = @options.parse
-        ::Rubycritic::CommandFactory.create(parsed_options).execute
+        result = ::Rubycritic::CommandFactory.create(parsed_options).execute
```

Comment:
I believe "execute" should rather return a kind of Reporter initialized with the given options from the options_parser. And this reporter you can then ask: 

* "Is my score to low?" (with the reporter knowing if the flag has been set or not)
* "What should I return?"

You could check out how we do it in [`Reek`](https://github.com/troessner/reek/blob/master/lib/reek/cli/application.rb#L31) - as you can see, the "Application" still has access to the corresponding exit code even at the very end of the life time.

I'm thinking about something along the lines (Pseudo code obviously):

```Ruby
reporter = ::Rubycritic::CommandFactory.create(parsed_options).execute
reporter.exit_status
```

With the reporter knowing the options and then encapsulating this logic.